pyccel/parser/semantic.py

```diff
@@ -443,15 +443,20 @@ def get_function(self, name):
 
 
     def get_import(self, name):
-        """."""
+        """
+        Search for an import with the given name in the current namespace.
+        Return None if not found.
+        """
 
         imp = None
-
         container = self.namespace
         while container:
 
-            if name in container.imports['imports']:
-                imp =  container.imports['imports'][name]
+            for key,_ in container.imports['imports'].items():
+                if key == name:
+                    imp = container.imports['imports'][key]
+
+            if imp:
```

Comment:
Why do you need 2 ifs? Can the expression above return None?

Reply:
yes, when the import is not in the given namespaces.
here the dict is orderedDict and I need the last import 
example:
```PYTHON
import numpy as np
import math as np
x = np.sqrt(x) #here sqrt is from math math 
```
so I found a un imp in the current scop no need to search in parents cop

Reply:
I meant can `container.imports['imports'][key]` return None? If not then can't the break go next to the assignment

Reply:
first container.imports['imports'][key] cant return None 
Second imp is initialized with None, so if the condition name == key doesn't occur. imp is still None and we need to search in the parent scope 
otherwise, we need to break.

Reply:
I think I'm not being clear. My suggested code is:
```
           for key,_ in container.imports['imports'].items():
                if key == name:
                    imp = container.imports['imports'][key]
                    break
```
If `container.imports['imports'][key]` can't return None then this should be possible and it saves an if statement

Reply:
this break will only  stop the for loop, not the while loop  
I need to leave the while loop when I find an import because there is no need to search in the parent scope

Reply:
Of course! Sorry. This should be possible if the discussion below removes the for loop though